packages/web3-eth-contract/src/contract.ts

```diff
@@ -132,17 +172,132 @@ export class Contract<Abi extends ContractAbi>
 		this._jsonInterface = [...result] as unknown as Abi;
 	}
 
-	// eslint-disable-next-line class-methods-use-this
-	private _createContractFunction(
-		name: string,
-		// eslint-disable-next-line @typescript-eslint/no-unused-vars
-		_abi: AbiFunctionFragment,
-	): ContractBoundFunction {
-		return name;
+	private _createContractMethod(abi: AbiFunctionFragment): ContractBoundMethod {
+		return (...params) => {
+			validator.validate(abi.inputs, params);
+
+			if (abi.stateMutability === 'payable' || abi.stateMutability === 'pure') {
+				return {
+					arguments: params,
+					call: async (options?: PayableCallOptions, block?: BlockNumberOrTag) =>
+						this._contractMethodCall(abi, params, options, block),
+					// TODO: Use `web3-eth-tx` package to return `PromiEvent` instead.
+					send: async (options?: PayableCallOptions) =>
+						this._contractMethodSend(abi, params, options),
+					estimateGas: async (options?: PayableCallOptions, block?: BlockNumberOrTag) =>
+						this._contractMethodEstimateGas(abi, params, options, block),
+					encodeABI: () => encodeMethodABI(abi, params),
+				} as unknown as PayableMethodObject<unknown, unknown>;
+			}
+
+			return {
+				arguments: params,
+				call: async (options?: NonPayableCallOptions, block?: BlockNumberOrTag) =>
+					this._contractMethodCall(abi, params, options, block),
+				send: async (options?: NonPayableCallOptions) =>
+					this._contractMethodSend(abi, params, options),
+				estimateGas: async (options?: NonPayableCallOptions, block?: BlockNumberOrTag) =>
+					this._contractMethodEstimateGas(abi, params, options, block),
+				encodeABI: () => encodeMethodABI(abi, params),
+			} as unknown as NonPayableMethodObject<unknown, unknown>;
+		};
+	}
+
+	private async _contractMethodCall<Options extends PayableCallOptions | NonPayableCallOptions>(
+		abi: AbiFunctionFragment,
+		params: unknown[],
+		options?: Options,
+		block?: BlockNumberOrTag,
+	) {
+		return decodeMethodReturn(
+			abi,
+			await this.requestManager.send({
+				method: 'eth_call',
+				params: [
+					getTxCallParams({
+						abi,
+						params,
+						options,
+						contractOptions: this.options,
+					}),
+					block ?? BlockTags.LATEST,
+				],
+			}),
+		);
+	}
+
+	private async _contractMethodSend<Options extends PayableCallOptions | NonPayableCallOptions>(
+		abi: AbiFunctionFragment,
+		params: unknown[],
+		options?: Options,
+	) {
+		return decodeMethodReturn(
+			abi,
+			await this.requestManager.send({
+				method: 'eth_sendTransaction',
+				params: [
+					getSendTxParams({
+						abi,
+						params,
+						options,
+						contractOptions: this.options,
+					}),
+				],
+			}),
+		);
+	}
+
+	private async _contractMethodEstimateGas<
+		Options extends PayableCallOptions | NonPayableCallOptions,
+	>(abi: AbiFunctionFragment, params: unknown[], options?: Options, block?: BlockNumberOrTag) {
+		return hexToNumber(
+			await this.requestManager.send({
+				method: 'eth_estimateGas',
```

Comment:
Should be using `web3Eth.estimateGas`

Reply:
Yes and no both. I want to avoid the circular dependency. 

If `web3-eth` is using `web3-eth-contract` package then we can't use the other way around. To provide `web3.eth.Contract` interface to users that dependency is required. 

Suggest if you have any other workaround?

Reply:
I'd like to avoid having multiple implementations of the RPC methods, so we don't repeat the mistakes of `1.x`. I guess that means splitting the rpc wrappers into a separate package (`web3-eth-rpc-methods`)?

Reply:
I think `web3-eth-tx` may solve this issue by having common code that can avoid the circular dependency between `web3-eth` and `web3-eth-contract` package.

Reply:
`web3-eth-tx` doesn't exist anymor. The tx util functions also needed access to `Web3Eth` methods, so it was made apart of `Web3Eth` for the same circular dependency issues

Reply:
In that case, I don't see any other way to have that small duplicate code to avoid circular dependency. Maybe as things shape up we may find some other solution. If you think of any other way around, please share.